### PR TITLE
DATES-615 - Use annotated field name on repository order by clause.

### PR DESCRIPTION
**Pull request by sothawo**
This fixes that in repository queries the name defined in a `@Field(name="foo")` annotation was not considered in an order by clause.

* `ElasticsearchPartQuery` and `ElasticsearchQueryCreator` changes are handling a query with no criteria like _findAllByOrderByName()_ which up to now causes a NPE
* relevant implementation in `ElasticsearchTemplate` and `ElasticsearchRestTemplate`
* the rest is reworking the `QueryKeywordsTests` test class that now is abstract and has two implementations setting up the different Templates. Moved to Java base configuration there instead of the xml-defined test context

- [x] You have read the [Spring Data contribution guidelines](https://github.com/spring-projects/spring-data-build/blob/master/CONTRIBUTING.adoc).
- [x] There is a ticket in the bug tracker for the project in our [JIRA](https://jira.spring.io/browse/DATAES).
- [x] You use the code formatters provided [here](https://github.com/spring-projects/spring-data-build/tree/master/etc/ide) and have them applied to your changes. Don’t submit any formatting related changes.
- [x] You submit test cases (unit or integration tests) that back your changes.
- [x] You added yourself as author in the headers of the classes you touched. Amend the date range in the Apache license header if needed. For new types, add the license header (copy from another file and set the current year only).

### CODE REVIEW

File: src/test/java/org/springframework/data/elasticsearch/repository/query/keywords/QueryKeywordsRepositoryTests.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import org.springframework.data.elasticsearch.repository.config.EnableElasticsearchRepositories;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+/**
+ * {@link QueryKeywordsTests} using a Repository backed by an ElasticsearchTemplate.
+ *
+ * @author Peter-Josef Meisch
+ */
+@RunWith(SpringJUnit4ClassRunner.class)
```
Comment (mp911de) [inline src/test/java/org/springframework/data/elasticsearch/repository/query/keywords/QueryKeywordsRepositoryTests.java]: `@RunWith(SpringRunner.class)`

File: src/test/java/org/springframework/data/elasticsearch/repository/query/keywords/QueryKeywordsRepositoryTests.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+@ContextConfiguration(classes = {QueryKeywordsRepositoryTests.class, ElasticsearchConfiguration.class})
+@Configuration
+@EnableElasticsearchRepositories(considerNestedRepositories = true)
+public class QueryKeywordsRepositoryTests extends QueryKeywordsTests {
+
+	@ClassRule public static NodeResource nodeResource = new NodeResource();
```
Comment (mp911de) [inline src/test/java/org/springframework/data/elasticsearch/repository/query/keywords/QueryKeywordsRepositoryTests.java]: Nit: Something's odd with formatting (tabs vs. spaces?)

File: src/test/java/org/springframework/data/elasticsearch/repository/query/keywords/QueryKeywordsRestRepositoryTests.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+import org.springframework.data.elasticsearch.repository.config.EnableElasticsearchRepositories;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+/**
+ * {@link QueryKeywordsTests} using a Repository backed by an ElasticsearchRestTemplate.
+ *
+ * @author Peter-Josef Meisch
+ */
+@RunWith(SpringJUnit4ClassRunner.class)
```
Comment (mp911de) [inline src/test/java/org/springframework/data/elasticsearch/repository/query/keywords/QueryKeywordsRestRepositoryTests.java]: `@RunWith(SpringRunner.class)`

File: src/test/java/org/springframework/data/elasticsearch/repository/query/keywords/QueryKeywordsTests.java [hunk 4]
```diff
@@ -26,45 +26,48 @@
...
 
 		IndexInitializer.init(elasticsearchTemplate, Product.class);
 
-		repository.saveAll(
-				Arrays.asList(Product.builder().id("1").name("Sugar").text("Cane sugar").price(1.0f).available(false).build(),
-						Product.builder().id("2").name("Sugar").text("Cane sugar").price(1.2f).available(true).build(),
-						Product.builder().id("3").name("Sugar").text("Beet sugar").price(1.1f).available(true).build(),
-						Product.builder().id("4").name("Salt").text("Rock salt").price(1.9f).available(true).build(),
-						Product.builder().id("5").name("Salt").text("Sea salt").price(2.1f).available(false).build()));
+		repository.saveAll(Arrays.asList(
```
Comment (mp911de) [inline src/test/java/org/springframework/data/elasticsearch/repository/query/keywords/QueryKeywordsTests.java]: Nit: Extract product objects for readability into variables.

File: src/test/java/org/springframework/data/elasticsearch/repository/query/keywords/QueryKeywordsTests.java [hunk 5]
```diff
@@ -155,6 +158,40 @@ public void shouldSupportLessThanAndGreaterThan() {
 		assertThat(repository.findByPriceGreaterThanEqual(1.9f)).hasSize(2);
 	}
 
+	@Test // DATAES-615
+	public void shouldSupportSortOnStandardFieldWithCriteria() {
+		final List<String> sortedIds = repository.findAllByNameOrderByText("Salt").stream() //
```
Comment (mp911de) [inline src/test/java/org/springframework/data/elasticsearch/repository/query/keywords/QueryKeywordsTests.java]: Nit: No need for `final` keyword in local variables.

File: src/test/java/org/springframework/data/elasticsearch/repository/query/keywords/QueryKeywordsTests.java (hunk 4 above)
Comment (sothawo) [reply src/test/java/org/springframework/data/elasticsearch/repository/query/keywords/QueryKeywordsTests.java]: done

File: src/test/java/org/springframework/data/elasticsearch/repository/query/keywords/QueryKeywordsTests.java (hunk 5 above)
Comment (sothawo) [reply src/test/java/org/springframework/data/elasticsearch/repository/query/keywords/QueryKeywordsTests.java]: that's IJ on refcatoring - must check where to disable that

**Pull request merged**